network/src/primary.rs

```diff
@@ -164,29 +167,30 @@ impl PrimaryNetwork {
 
 #[derive(Default)]
 pub struct PrimaryToWorkerNetwork {
-    clients: HashMap<SocketAddr, PrimaryToWorkerClient<Channel>>,
+    clients: HashMap<Multiaddr, PrimaryToWorkerClient<Channel>>,
+    config: mysten_network::config::Config,
 }
 
 impl PrimaryToWorkerNetwork {
-    fn client(&mut self, address: SocketAddr) -> PrimaryToWorkerClient<Channel> {
+    fn client(&mut self, address: Multiaddr) -> PrimaryToWorkerClient<Channel> {
         self.clients
-            .entry(address)
-            .or_insert_with(|| Self::create_client(address))
+            .entry(address.clone())
+            .or_insert_with(|| Self::create_client(&self.config, address))
             .clone()
     }
 
-    fn create_client(address: SocketAddr) -> PrimaryToWorkerClient<Channel> {
-        // TODO use TLS
```

Comment:
Ditto: I'd like to not lose this comment, because in the mTLS context, using TLS is not just about a change of protocol, but requires configuring a custom verifier.